drivers/sensor/bma421/bma421.c

```diff
@@ -143,85 +167,60 @@ static const struct sensor_driver_api bma421_driver_api = {
 	.channel_get = bma421_channel_get,
 };
 
-
-
-
-
-int bma421_init(struct device *dev)
+int bma421_init_driver(struct device *dev)
 {
 	struct bma421_data *drv_data = dev->driver_data;
-	u8_t id = 0U;
-	drv_data->i2c = device_get_binding(CONFIG_BMA421_I2C_MASTER_DEV_NAME);
+	uint8_t res;
+
+	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
 	if (drv_data->i2c == NULL) {
 		LOG_DBG("Could not get pointer to %s device",
-				CONFIG_BMA421_I2C_MASTER_DEV_NAME);
+			    DT_INST_BUS_LABEL(0));
 		return -EINVAL;
 	}
 
-	/* read device ID */
-	if (i2c_reg_read_byte(drv_data->i2c, BMA421_I2C_ADDRESS,
-				BMA421_REG_CHIP_ID, &id) < 0) {
-		LOG_DBG("Could not read chip id");
-		return -EIO;
-	}
+	memset(&drv_data->dev, 0, sizeof(struct bma4_dev));
+	drv_data->dev.intf = BMA4_I2C_INTF;
+	drv_data->dev.intf_ptr = drv_data;
+	drv_data->dev.variant = BMA42X_VARIANT;
+	drv_data->dev.bus_read = bm421_i2c_read;
+	drv_data->dev.bus_write = bm421_i2c_write;
+	drv_data->dev.delay_us = bma421_delay_us;
+	drv_data->dev.read_write_len = 8;
+	drv_data->dev.resolution = BMA4_16_BIT_RESOLUTION;
 
-	if (id != BMA421_CHIP_ID) {
-		LOG_DBG("Unexpected chip id (%x)", id);
-		return -EIO;
-	}
+	res = bma421_init(&drv_data->dev);
+	if (res)
+		LOG_ERR("init failed err %d", res);
```

Comment:
Zephyr's [coding style](https://docs.zephyrproject.org/latest/contribute/index.html#coding-style) state that one must add "braces to every if, else, do, while, for and switch body".

Reply:
oups. fixed